ch_tools/chadmin/cli/wait_group.py

```diff
@@ -98,28 +98,31 @@ def wait_replication_sync_command(
     start_time = time.time()
     deadline = start_time + total_timeout.total_seconds()
 
-    # Sync tables in cycle
-    for replica in list_table_replicas(ctx):
-        full_name = f"`{replica['database']}`.`{replica['table']}`"
-        time_left = deadline - time.time()
-        timeout = min(replica_timeout.total_seconds(), time_left)
+    try:
+        # Sync tables in cycle
+        for replica in list_table_replicas(ctx):
+            full_name = f"`{replica['database']}`.`{replica['table']}`"
+            time_left = deadline - time.time()
+            timeout = min(replica_timeout.total_seconds(), time_left)
 
-        try:
             execute_query(
                 ctx,
                 f"SYSTEM SYNC REPLICA {full_name}",
                 format_=None,
                 timeout=timeout,
                 settings={"receive_timeout": timeout},
             )
-        except ReadTimeout:
-            logging.error("Timeout while running SYNC REPLICA on {}.", full_name)
+    except requests.exceptions.ReadTimeout:
+        logging.error("Read timeout while running query.")
+        sys.exit(1)
+    except requests.exceptions.ConnectionError:
+        logging.error("Connection error while running query.")
+        sys.exit(1)
```

Comment:
> Why we should exit here? Is reraise the exception + logging.error or raise new exception with custom msg not enough?
That is good point - what do you think about fix problem globally here - https://github.com/yandex/ch-tools/blob/main/ch_tools/chadmin/cli/chadmin_group.py#L54
Just adding `raise` in except branch. I afraid it is initial problem. And not only for `wait replication-sync` problem.

> Also maybe for `requests.exceptions.ConnectionError` add retries? Because we can have some short network issues
It is already here - https://github.com/yandex/ch-tools/blob/main/ch_tools/common/clickhouse/client/clickhouse_client.py#L165

Reply:
Yeah, we need to review all such emergency exists, we got a lot of them:
```
./ch_tools/s3_credentials/main.py:        sys.exit(1)
./ch_tools/s3_credentials/main.py:        sys.exit(1)
./ch_tools/s3_credentials/main.py:    sys.exit(0)
./ch_tools/monrun_checks/main.py:            sys.exit(1)
./ch_tools/monrun_checks/main.py:                sys.exit(1)
./ch_tools/chadmin/cli/wait_group.py:            sys.exit(1)
./ch_tools/chadmin/cli/wait_group.py:                sys.exit(1)
./ch_tools/chadmin/cli/wait_group.py:            sys.exit(0)
./ch_tools/chadmin/cli/wait_group.py:    sys.exit(1)
./ch_tools/chadmin/cli/wait_group.py:        sys.exit(0)
./ch_tools/chadmin/cli/wait_group.py:    sys.exit(1)
./ch_tools/chadmin/cli/zookeeper_group.py:    sys.exit(1)
```
Let's deal with it in the separate pr

>It is already here 
Thanks, missed it.